WLC/web_endpoint.py

```diff
@@ -31,10 +31,18 @@ def api_upload_image():
         pic = Picture(img, 0, 0, width, height, None)
         saved, key = save_image_to_azure('pictures', pic.get_image())
 
-        code, fixed_code, result, error = CodeExecutor().execute_code_img(pic)
+        executor = CodeExecutor()
+        code, fixed_code, result, error = executor.execute_code_img(pic)
+
+        if 'test_key' in request.json:
+            hacker_rank_results = executor.execute_hacker_rank(code, request.json.get('test_key'))
+        else:
+            hacker_rank_results = []
+
         ar = _get_ar_coordinates(pic, error)
 
-        response = {'unfixed': code, 'fixed': fixed_code, 'result': str(result), 'error': str(error), 'key': key, 'ar': ar}
+        response = {'unfixed': code, 'fixed': fixed_code, 'result': str(result), 'error': str(error), 'key': key,
+                    'ar': ar, 'hacker_rank_results': hacker_rank_results}
```

Comment:
How about we just call it `test_results` or something unrelated to HackerRank? It's because you said the hackerrank doesn't really provide anything special so if we decide to change the test runner, it will introduce minimal changes.